### ISSUE TITLE
Best model ist not saved correctly

### ISSUE THREAD

**Issue by DavidBaldsiefen** [ultralytics/yolov5#6778]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

I have performed a training on my custom dataset with `patience 100` using the following command:

`train.py --epochs 1000 --data ../data.yaml --weights yolov5m.pt --batch 8 --cache`

Training stopped after 677 epochs with the best epoch being observed at epoch 576 with mAP50=0.9398 and mAP=0.6629.
This data can be found both in the `results.csv` as well as in wandb.

However, when I manually run validation on the same set, I get the following results: mAP50=0.941 mAP=0.656

`python3 val.py --weights runs/train/exp21/weights/best.pt --task val --data ../data.yaml`

As you can see, based on fitness, best.pt clearly performs worse than the best epoch during training did. How can this be explained? I have double checked the directory and values

### Environment

- Latest YOLOv5
- Ubuntu 18.04

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@DavidBaldsiefen all models are saved correctly. val.py will use slightly different inference settings when called directly than when called during training. These differences, i.e. specifically 0.5 padding lead to improvements in common datasets like COCO at the expense of some speed. See val.py calls in train.py vs val.py argparser arguments (and dataloader which val.py creates vs valloader which train.py creates):

https://github.com/ultralytics/yolov5/blob/cea994b3f62a55106c3b1404f36e8aa780f324e0/train.py#L369-L380

https://github.com/ultralytics/yolov5/blob/cea994b3f62a55106c3b1404f36e8aa780f324e0/val.py#L319-L348
